cli/cli.py

```diff
@@ -0,0 +1,258 @@
+        try:
+            parsed_line = parse_access_log_line(line)
+
+            # Empty dict means line was filtered out by URL.
+            if not parsed_line:
+                stats[LINES_FILTERED] += 1
+                continue
+
+            if suppress_unspecified:
+                if parsed_line.get(EVENT_SIGNATURE) == DEFAULT_EVENT[EVENT_SIGNATURE]:
+                    stats[EVENTS_SUPPRESSED] += 1
+                    continue
+
+            cef_event = write_cef_event(parsed_line)
+            logger.info(cef_event)
+            stats[EVENTS_WRITTEN] += 1
+        except (DateParseError, LogParseError) as err:
+            errors.append(err)
+    return (stats, errors)
+
+
+@click.group()
+def auditmatica():
+    """Auditmatica: Archivematica auditing package"""
+    pass
+
+
+@auditmatica.command("write-cef")
+@click.argument("log", type=click.File("r", encoding="utf-8"), default=sys.stdin)
+@click.option(
+    "-o",
+    "--output",
+    type=click.Path(),
+    default=None,
+    help="Filepath for output CEF file (default=None, print to stdout)",
+)
+@click.option(
+    "-s", "--syslog", is_flag=True, help="Write CEF events to syslog instead of file"
+)
+@click.option(
+    "--syslog-address",
+    type=str,
+    default="/dev/log",
+    help="Address for syslog connection (default='/dev/log')",
+)
+@click.option(
+    "--syslog-facility",
+    type=str,
+    default="USER",
+    help="Facility for syslog messages (default='USER')",
+)
+@click.option(
+    "--syslog-port", type=int, default=None, help="Port for remote syslog connections"
+)
+@click.option(
+    "--ss-base-url",
+    type=str,
+    default="http://127.0.0.1:62081",
+    help="Base URL for Archivematica Storage Service (default='http://127.0.0.1:62081')",
+)
+@click.option(
+    "--suppress",
+    is_flag=True,
+    help="Suppress log lines that do not map to a specific event instead of reverting to default event",
+)
+@click.option(
+    "-v", "--verbose", is_flag=True, help="Enable verbose error message reporting"
+)
+def write_cef_events(
+    log,
+    output,
+    syslog,
+    syslog_address,
+    syslog_facility,
+    syslog_port,
+    ss_base_url,
+    suppress,
+    verbose,
+):
+    """Write Common Event Format (CEF) log from nginx access log."""
+    log_lines = log.readlines()
+
+    _configure_logging(syslog, syslog_address, syslog_facility, syslog_port, output)
+
+    (stats, errors) = _process_log(log_lines, ss_base_url, suppress)
+
+    # Print user-facing messages to stderr.
+    click.echo("Done!", err=True)
+    click.echo(f"CEF events written: {stats[EVENTS_WRITTEN]}", err=True)
+    if suppress:
+        click.echo(f"Default events suppressed: {stats[EVENTS_SUPPRESSED]}", err=True)
+    click.echo(f"Log lines filtered out by URL: {stats[LINES_FILTERED]}", err=True)
+    if errors:
+        click.echo(f"Errors: {len(errors)}", err=True)
+        if verbose:
+            for error in errors:
+                click.echo(error, err=True)
```

Comment:
```suggestion
                click.echo(str(error).rstrip("\n), err=True)
```
I believe these strings arriving here will already be newline terminated and so we can save space by stripping that. Echo will otherwise add another.